src/datachain/catalog/catalog.py

```diff
@@ -1120,6 +1120,9 @@ def get_dataset_with_remote_fallback(
         pull_dataset: bool = False,
         update: bool = False,
     ) -> DatasetRecord:
+        if version:
+            update = False
```

Comment:
Changing the user argument doesn't seem like a good idea. We could either raise error or assert that having both version and update to true is not advisable.

Still, how would this work if we specify version>=0.2 as spec and update? Shouldn't this still update the version?

Reply:
Then why not adding assertion that `version` and `update` should both not be provided or assign it to a different argument instead of modifying the function argument?

Reply:
the can be provided - that's fine, we just ignore it then (from the upstream sometimes we want to pass exact version and don't want to track update flag). It is fine that we just ignore it here I think.

The same btw for the users:

```
dc.read_dataset("some.remote.dataset", version="1.0.0", update=True)
```

we just ignore the update flag. Is it the best semantics - don't know, but seems reasonable. Let me know WDYT.

Reply:
Semantics feels little weird tbh. But dismissing the change request to move forward